plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -310,8 +293,54 @@ func init() {
 		}
 		return &OpenWeatherMap{
 			ResponseTimeout: tmout,
-			Units:           defaultUnits,
-			BaseUrl:         defaultBaseURL,
+			BaseUrl:         defaultBaseUrl,
 		}
 	})
 }
+
+func (n *OpenWeatherMap) Init() error {
+	var err error
+	n.baseUrl, err = url.Parse(n.BaseUrl)
+	if err != nil {
+		return err
+	}
+
+	// Create an HTTP client that is re-used for each
+	// collection interval
+	n.client, err = n.createHttpClient()
+	if err != nil {
+		return err
+	}
+
+	switch n.Units {
+	case "imperial", "standard", "metric":
+	case "":
+		n.Units = defaultUnits
+	default:
+		return fmt.Errorf("unknown units: %s", n.Units)
+	}
+
+	return nil
+}
+
+func (n *OpenWeatherMap) formatURL(path string, city string) string {
+	v := url.Values{
+		"id":    []string{city},
+		"APPID": []string{n.AppId},
+	}
+
+	if n.Units != "" {
+		v["units"] = []string{n.Units}
+	}
+
+	if n.Lang != "" {
```

Comment:
Do this in one of your init functions (set a default in `init`, or check and set in `Init`)

Reply:
done